src/libs/geolocation.js

```diff
@@ -10,20 +11,24 @@ const geolocationPermissions = {
 let hasPermissionModalOpenedOnce = false;
 
 export async function showGeolocationModalIfNeeded() {
+  if (!isMobileDevice()) {
```

Comment:
You're doing this test in DirectionPanel, it can be removed from here to focus only on the permission stuff.